content/docs/reference/tutorials/kubernetes/tutorial-migrate-nodegroups.md

```diff
@@ -0,0 +1,454 @@
+Next, we'll create a new node group in AWS using Pulumi for the `4xlarge` node
+group. This is as simple as defining a new node group in our program:
+
+<p align="center"><img src="/images/docs/reference/kubernetes/ng-4xlarge.svg" width="650"/></p>
+
+You can make this change yourself in your `index.ts` by running the following: 
+
+```bash
+cp steps/step1/index.ts index.ts
+pulumi up
+```
+
+Once the update is complete, verify the new `c5.4xlarge` node group is up and running:
+
+```bash
+kubectl get nodes -o wide --show-labels -l beta.kubernetes.io/instance-type=c5.4xlarge
+```
+
+### Step 2: Retargeting NGINX at the `4xlarge` Node Group
+
+Now, we'll retarget the NGINX Service away from the `2xlarge` node group over to the
+`4xlarge` node group, by changing its node selector scheduling terms:
+
+<p align="center"><img src="/images/docs/reference/kubernetes/target-ng-4xlarge.svg" width="650"/></p>
+
+You can make this change yourself in your `index.ts` by running the following: 
+
+```bash
+cp steps/step2/index.ts index.ts
+pulumi up
+```
+
+> **Note:** Given that the termination process must gracefully shutdown, and process
+all in-flight requests, this may take a few minutes to complete.
+
+Once the update is complete, verify the NGINX pods are now running in the new
+`4xlarge` node group by retrieving its Pods, the `c5.4xlarge` nodes, and
+noticing that the Pods are in fact running on updated nodes.
+
+```bash
+kubectl get pods --all-namespaces -o wide --show-labels -l app=nginx-ing-cntlr
+kubectl get nodes -o wide --show-labels -l beta.kubernetes.io/instance-type=c5.4xlarge
+```
+
+You should also notice a linear up-tick in **requests per second** in the
+load testing results, due to the more capable `c5.4xlarge` worker instances
+being used.
+
+### Step 3: Decomissioning the `2xlarge` Node Group
+
+With NGINX validated to be up and running on the `4xlarge` node group, we can
+now commence the decomissioning of the original `2xlarge` node group no
+longer in use.
+
+First, we'll instruct the Kubernetes APIServer to drain the `2xlarge` node group using
+`kubectl drain`. We've added a helper script to facilitate this process:
+
+```bash
+./scripts/drain-t3.2xlarge-nodes.sh
+```
+
+Once draining is complete, let's instruct the Kubernetes APIServer to delete
+and remove the nodes from the cluster using `kubectl delete node`:
+
+```bash
+./scripts/delete-t3.2xlarge-nodes.sh
+```
+
+After node deletion from the cluster has completed, and we've verified that the
+instances backing our load balancer are up-to-date and no longer include the
+`2xlarge` node group, we can remove its definition from our Pulumi program:
+
+<p align="center"><img src="/images/docs/reference/kubernetes/remove-ng-2xlarge.svg" width="650"/></p>
+
+You can make this change yourself in your `index.ts` by running the following: 
+
+```bash
+cp steps/step3/index.ts index.ts
+pulumi up
+```
+
+If we've executed all of the steps as follows, then we'll have successfully migrated
```

Comment:
Should it be "If you've executed all of the previous steps"

Reply:
Done